lib/tasks/seek_upgrades.rake

```diff
@@ -125,7 +126,22 @@ namespace :seek do
         counter += 1
       end
     end
-    puts "...Added policies to #{counter} sample types"
+    puts "... Added policies to #{counter} sample types"
+  end
+
+  task(fix_previous_sample_type_permissions: [:environment]) do
+    only_once('fix_previous_sample_type_permissions 1.16.0') do
+      puts '... Updating previous sample type permissions ...'
+      SampleType.includes(:policy).where.not(policy_id: nil).each do |sample_type|
+        policy = sample_type.policy
+        if policy.access_type == Policy::VISIBLE
+          policy.update_column(:access_type, Policy::ACCESSIBLE)
```

Comment:
Again a question. Does a sample type need to be accessible in order to be able to download the template?

Reply:
Yes, because it goes throught the content blob controller it checks the permissions on the associated asset. There is an example here: https://testing.sysmo-db.org/sample_types/97